fuel_utility.py

```diff
@@ -20,89 +20,114 @@
 from docopt import docopt
 import zipfile
 import io
-import xml.etree.ElementTree as ET
-import xml.dom.minidom
+from lxml import etree as ET
 import os.path
 
-FUEL_URI="https://fuel.ignitionrobotics.org/1.0/OpenRobotics/models"
-WAIT_BETWEEN_DOWNLOADS_IN_SECONDS=5
+FUEL_URI = 'https://fuel.ignitionrobotics.org/1.0/OpenRobotics/models'
+WAIT_BETWEEN_DOWNLOADS_IN_SECONDS = 5
 
-class FuelModelUtility: 
+class FuelModelUtility:
 
-    models = []  
+    models = []
 
     def __init__(self):
-        logging.info("Starting fuel models..")
+        logging.info('Starting fuel models..')
 
     def getModelList(self):
         return self.models
 
     def appendModelSearchResults(self, page, category):
-        response = requests.get(FUEL_URI+"?q="+category+"&page="+str(page))
+        url = '%s?q=%s&page=%s' % (FUEL_URI, category, str(page))
+        response = requests.get(url)
         modelArray = json.loads(response.text)
         self.models.extend(modelArray)
         page += 1
-        if (len(modelArray)>=20):
+        if (len(modelArray) >= 20):
             self.appendModelSearchResults(page, category)
 
-    def appendSingleModel(self, model_name):
-        response = requests.get(FUEL_URI+"/"+model_name)
+    def appendSingleModel(self, model_info):
+        if ('=' in model_info):
+            model_name, model_version = model_info.split('=')
```

Comment:
it does make sense to specify each model version number separately as that's how fuel model seem to be updated (asynchronously from each other)..
a potential alternative could also be that we always fetch the latest version (2 in this case) for each model first and in case there is no response from the URL, fetch the older version.. upside being we don't keep track of individual model numbers but downside being there might be a lot of retries making it slower

Reply:
In my understanding, some models are at version 3 or 4. Newer models are still at version 1. While there does seem to be a lot of models where version 2 seems to be the latest version available, I don’t think there is any discernible standard here. Therefore, in my view having the version be defined at the model layer was better. Since there is an API call that is executed before the download to capture metadata, the best approach here is to select the latest version by default. 

Example: https://fuel.ignitionrobotics.org/1.0/OpenRobotics/models/VisitorChair
‘version:”2”’

Will put that in!